### ISSUE TITLE
bug with, or confused about, mysql::client

### ISSUE THREAD

**Issue by kernelsmith** [TalkingQuickly/rails-server-template#47]
TL;DR When bootstrapping this node I get mysql::client cookbook does not exist, stemming from https://github.com/TalkingQuickly/rails-server-template/blob/master/roles/mysql-server.json#L23

I'm running w/a couple differences:
1.  Using chef server, but I just pushed everything up including all the berks-installed cookbooks etc (including the mysql cookbook)
2.  Not using chef-solo-search.  

When bootstrapping this node I get mysql::client cookbook does not exist.  Looking into this, I can't see how it could possibly work correctly.  The node definition (https://github.com/TalkingQuickly/rails-server-template/blob/master/nodes/rails-postgres-redis1.json#L64) references the role, the role definition (https://github.com/TalkingQuickly/rails-server-template/blob/master/roles/mysql-server.json#L23) references the mysql::client recipe, however the mysql cookbook defined in the berksfile doesn't have any recipes, rather it's a "library cookbook that provides resource primitives (LWRPs) for use in recipes. It is designed to be a reference example for creating highly reusable cross-platform cookbooks." (https://github.com/chef-cookbooks/mysql).  It doesn't have any of it's own recipes, certainly none called client.  Grepping ~/.berkshelf/cookbooks and this repo for "depends 'mysql'" only reveals a few tests that depend on it, but no actual cookbooks.

I feel like I'm missing a wrapper cookbook or the mysql cookbook used to have actual recipes in it and was switched out for an LWRP lib cookbook.  What am I missing?

env info:

```
$ chef -v
Chef Development Kit Version: 0.10.0
chef-client version: 12.5.1
berks version: 4.0.1
kitchen version: 1.4.2
```

**Comment by kernelsmith** [comment]
I notice the mysql cookbook has client and server recipes but only on an older tag branch: https://github.com/chef-cookbooks/mysql/blob/v5.5.1/recipes/client.rb  These files don't exist in master

**Comment by kernelsmith** [comment]
oh, ok, you're Berksfile.lock has you on an older version of the mysql cookbook, and everything in the 5.x branch/tag seems to have the recipes.  Should the berksfile have a version constraint then?

**Comment by kernelsmith** [comment]
Seems like maybe 'mysql', '<6.0.0' maybe?

**Comment by kernelsmith** [comment]
https://github.com/chef-cookbooks/mysql/blob/v6.0.0/CHANGELOG.md yeah, 6.0.0 doesn't have recipes.  I think you should blow away your Berksfile.lock and try your build again, I think you'll find at least this one cookbook has moved beyond compatibility w/your setup.

**Comment by kernelsmith** [comment]
so I'm not sure how to fix it in the berksfile, but preserving the version of the mysql cookbook via the Berksfile.lock to anything in the 5.x range (I used 5.3.6 since that's what you had) works, but since 6.0+ are incompatible w/this setup (would req

**Comment by kernelsmith** [comment]
ok, AFAICT, you can't use the standard version constraints in the Berksfile w/the :git nomenclature.  You can specify a tag, branch, or commit like:

```
cookbook "mysql", git: "https://github.com/opscode-cookbooks/mysql.git", tag: "5.6.3"
```
Tho it would make far more sense if you could use the standard >, <, ~> etc approaches, otherwise you're tied to a specific version, vs say an entire major version

**Comment by kernelsmith** [comment]
regardless however, pinning in the Berksfile is definitely better than pinning in the Berksfile.lock

**Comment by TalkingQuickly** [comment]
Hi, You're 100% right about this, the latest version of the example configuration doesn't include MySQL because I haven't worked with it for rather a long time so don't have the knowledge needed to write about the best practices. I have got a note made to do a blog post over the next few months about creating a wrapper cookbook for MySQL!

**Comment by kernelsmith** [comment]
Hey what do you know for once!  ;) I have learned how to wrap these kinds of "new-fangled" cookbooks finally, tho I haven't wrapped this one.
